keps/sig-node/2371-cri-pod-container-stats/README.md

```diff
@@ -0,0 +1,734 @@
+    * Calls cAdvisor directly to obtain node, pod, and container stats
+* [CRI stats provider](https://github.com/kubernetes/kubernetes/blob/release-1.20/pkg/kubelet/stats/cri_stats_provider.go#L54)
+    * Calls CRI implementation to provide some minimum container level stats, but still relies on cAdvisor for the majority of container level stats + pod level stats + node level stats
+
+#### Current Fulfiller of Summary API Stats & Future Proposal 
+
+Below is a table describing which stats come from what source now, as well a proposal of which should come from where in the future. It also includes which fields roughly correspond to fields in the `/metrics/cadvisor` endpoint, some of which will not come from the CRI for the first iteration of this KEP. See more below.
+
+|Top level object              |`/stats/summary` Field|`/metrics/cadvisor` field                       |Level Needed in `/stats/summary`|Currently provided by:|Proposed to be provided by:|
+|------------------------------|----------------------|------------------------------------------------|--------------------------------|----------------------|---------------------------|
+|InterfaceStats (Network)      |RxBytes               |container_network_receive_bytes_total           |Pod                             |cAdvisor              |CRI                        |
+|                              |RxErrors              |container_network_receive_errors_total          |Pod                             |cAdvisor              |CRI                        |
+|                              |TxBytes               |container_network_transmit_bytes_total          |Pod                             |cAdvisor              |CRI                        |
+|                              |TxErrors              |container_network_transmit_errors_total         |Pod                             |cAdvisor              |CRI                        |
+|                              |N/A                   |container_network_receive_packets_dropped_total |N/A                             |cAdvisor              |CRI or N/A                 |
+|                              |N/A                   |container_network_receive_packets_total         |N/A                             |cAdvisor              |CRI or N/A                 |
+|                              |N/A                   |container_network_transmit_packets_dropped_total|N/A                             |cAdvisor              |CRI or N/A                 |
+|                              |N/A                   |container_network_transmit_packets_total        |N/A                             |cAdvisor              |CRI or N/A                 |
+|CPUStats                      |UsageNanoCores        |N/A                                             |Pod and Container               |cAdvisor              |CRI                        |
```

Comment:
I think it would make sense to omit metrics in CRI that are effectively derived from other metrics, or based on config values of the container.

Is `UsageNanoCores` derived from `UsageCoreNanoSeconds`? If so, can we have kubelet derive it itself, rather than getting it from CRI?

Reply:
by the looks of it, we omitted UsageNanoCores anyway. I'll update it to be `CRI or Kubelet`